library/alloc/src/string.rs

```diff
@@ -117,27 +117,85 @@ use crate::vec::Vec;
 ///
 /// # UTF-8
 ///
-/// `String`s are always valid UTF-8. This has a few implications, the first of
-/// which is that if you need a non-UTF-8 string, consider [`OsString`]. It is
-/// similar, but without the UTF-8 constraint. The second implication is that
-/// you cannot index into a `String`:
+/// `String`s are always valid UTF-8. If you need a non-UTF-8 string, consider
+/// [`OsString`]. It is similar, but without the UTF-8 constraint. Because UTF-8
+/// is a variable width encoding, `String`s can be smaller than an array of
```

Comment:
```suggestion
/// is a variable width encoding, `String`s are typically smaller than an array of
```

I think we want to push people towards String, and this seems pretty much always true -- for space-efficiency String is essentially always the better option.

Reply:
I think that's a great idea - 90f3f23495601a20bb04d74000b0a8e87d17444a